File: Practice/Others/2024 小红书/b.cpp
#include <bits/stdc++.h>

int main() {
	int n;
	std::cin >> n;
	std::vector<int> a(n + 1);
	for (int i = 1; i <= n; i++) {
		std::cin >> a[i];
	}
	int ans = 0;
	for (int i = 1; i <= n; i++) {
		int flag = 0;
		for (int j = i - 1; j >= 1; j--) {
			for (int k = j + 1; k < i; k++) {
				if (std::gcd(j, k) != 1) flag = 1;
			}
			if (flag) {
				ans = std::max(ans, i - j);
				break;
			}
		}
		if (!flag) {
			ans = std::max(ans, i);
		}
	}
	std::cout << ans << '\n';
}

File: Practice/Others/2024 小红书/b.py
import math

n = int(input())
a = [0] + list(map(int, input().split()))  # Read input and store it in a list
ans = 0
for i in range(1, n + 1):
    flag = False
    for j in range(i - 1, 0, -1):
        for k in range(j + 1, i):
            if math.gcd(j, k) != 1:
                flag = True
        if flag:
            ans = max(ans, i - j)
            break
    if not flag:
        ans = max(ans, i)
print(ans)

